### PR TITLE
spelling fixes for lib folder

### PR DESCRIPTION
**Pull request by h00die**
Cleanup of spelling across the `lib` folder. I avoided fixing apostrophes.

Most of these changes are in comments, some are in user prompts which will be nice.

## Verification

- [ ] manual review
- [ ] Start `msfconsole`
- [ ] use some things to make sure they dont break

### CODE REVIEW

**Comment by h00die** [comment]
`Authentification` seems to be a popular made up word, you can see it in a few libraries. I fixed some of them, but can get more aggressive. Wanted to call this out for discussion incase we actually want to revert the changes I made in here.

File: lib/rex/proto/drda/utils.rb [hunk 1]
```diff
@@ -111,7 +111,7 @@ def self._info_secchkrm(ddm)
         next
       end
     end
-    if info_hash[:serverity].to_i.zero? and info_hash[:security_check_code].to_i.zero?
+    if info_hash[:severity].to_i.zero? and info_hash[:security_check_code].to_i.zero?
```
Comment (h00die) [inline lib/rex/proto/drda/utils.rb:114]: Someone should manually review this, I didn't see when the hash was created any reference to `severity` or `serverity`. My guess is its been broken forever, but should def be reviewed.

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb [hunk 2]
```diff
@@ -513,7 +513,7 @@ def cmd_interact(*args)
 
   @@set_timeouts_opts = Rex::Parser::Arguments.new(
     '-c' => [true, 'Comms timeout (seconds)'],
-    '-x' => [true, 'Expiration timout (seconds)'],
+    '-x' => [true, 'Expiration timeout (seconds)'],
```
Comment (h00die) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:516]: https://github.com/rapid7/metasploit-framework/pull/18435#discussion_r1352045937

File: lib/rex/proto/drda/utils.rb (hunk 1 above)
Comment (adfoster-r7) [reply lib/rex/proto/drda/utils.rb:114]: It should be `severity_code` based on the assignment above https://github.com/rapid7/metasploit-framework/blob/ee3b8b40b690f6209cd3d8797eb26799e6e07565/lib/rex/proto/drda/utils.rb#L107

Comment (adfoster-r7) [reply lib/rex/proto/drda/utils.rb:114]:
```suggestion
    if info_hash[:severity_code].to_i.zero? and info_hash[:security_check_code].to_i.zero?
```

**Comment by adfoster-r7** [comment]
## Release Notes

Fix spelling mistakes in Metasploit's library folder

**Pull request merged**